django/db/models/sql/compiler.py

```diff
@@ -1911,18 +1910,6 @@ def execute_sql(self, returning_fields=None):
                     )
                 ]
                 cols = [field.get_col(opts.db_table) for field in self.returning_fields]
-            elif isinstance(opts.pk, CompositePrimaryKey):
-                returning_field = returning_fields[0]
-                cols = [returning_field.get_col(opts.db_table)]
-                rows = [
-                    (
-                        self.connection.ops.last_insert_id(
-                            cursor,
-                            opts.db_table,
-                            returning_field.column,
-                        ),
-                    )
-                ]
             else:
                 cols = [opts.pk.get_col(opts.db_table)]
```

Comment:
Small note that this is likely completely unnecessary for models that don't have an `AutoField` primary key and use `GeneratedField` (and thus have some members in `returning_fields`). I believe there's even a way to trigger a crash on Oracle since `last_insert_id` attempts to retrieve a sequence value and to have the improper value assigned to `GeneratedField` values on MySQL.

Reply:
See second commit about the actual crash this unveiled.